tests/storages_tests/test_with_server.py

```diff
@@ -30,13 +30,24 @@ def objective(trial: optuna.Trial) -> float:
     return f(x, y)
 
 
-def run_optimize(args: Tuple[str, str]) -> None:
-    study_name = args[0]
-    storage_url = args[1]
+def get_storage(storage_url: str, storage_mode: str) -> Union[str, BaseStorage]:
+    if storage_mode == "RDB":
+        return storage_url
+    elif storage_mode == "journal-redis":
+        journal_redis_storage = optuna.storages.JournalRedisStorage(storage_url)
+        return optuna.storages.JournalStorage(journal_redis_storage)
+    else:
+        assert False, f"The mode {storage_mode} is not supported."
+
+
+def run_optimize(study_name: str, storage_url: str, storage_mode: str, n_trials: int) -> None:
     # Create a study
-    study = optuna.load_study(study_name=study_name, storage=storage_url)
+    study = optuna.load_study(
+        study_name=study_name,
+        storage=get_storage(storage_url, storage_mode),
+    )
     # Run optimization
-    study.optimize(objective, n_trials=20)
+    study.optimize(objective, n_trials=n_trials)
 
 
 @pytest.fixture
```

Comment:
The journal-redis test will fail if executed multiple times. The reason seems to be that this function does not clean up the `journal-redis` database. Can you fix it?